src/rust/dns/lib.rs

```diff
@@ -0,0 +1,204 @@
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum DnsParserError {
+    #[error("Invalid hex string: {0}")]
+    InvalidHexString(String),
+    #[error("ParseInt error: {0}")]
+    ParseIntError(#[from] std::num::ParseIntError),
+    #[error("Invalid DNS response: {0}")]
+    InvalidDnsResponse(String),
+    #[error("unknown dns parser error")]
+    Unknown,
+}
+
+#[cxx::bridge(namespace = "workerd::rust::dns")]
+mod ffi {
+    /// CAA record representation
+    struct CaaRecord {
+        critical: u8,
+        field: String,
+        value: String,
+    }
+    /// NAPTR record representation
+    struct NaptrRecord {
+        flags: String,
+        service: String,
+        regexp: String,
+        replacement: String,
+        order: u32,
+        preference: u32,
+    }
+    extern "Rust" {
+        fn parse_caa_record(record: &str) -> Result<CaaRecord>;
+        fn parse_naptr_record(record: &str) -> Result<NaptrRecord>;
+    }
+}
+
+/// Given a vector of strings, converts each slice to UTF-8 from HEX.
+///
+/// # Errors
```

Comment:
I personally don't care about errors documentation much (panics are important). Is clippy requiring them? I thought we have it off. Feel free to drop if this is true.

Reply:
Yes, if the function is pub then it requires it.